libcore/FileUtils.vala

```diff
@@ -804,6 +804,178 @@ namespace PF.FileUtils {
             return -1;
         }
     }
+
+    internal string format_time (int seconds, out int time_unit) {
+        int minutes;
+        int hours;
+
+        if (seconds < 0) {
+            /* Just to make sure... */
+            seconds = 0;
+        }
+
+        if (seconds < 60) {
+            time_unit = seconds;
+
+            return ngettext ("%'d second", "%'d seconds", seconds).printf (seconds);
+        }
+
+        minutes = seconds / 60;
+        if (minutes < 60) {
+            time_unit = minutes;
+            return ngettext ("%'d minute", "%'d minutes", minutes).printf (minutes);
+        }
+
+        hours = minutes / 60;
+        if (hours < 4) {
+            time_unit = minutes + hours;
+            return _("%s, %s").printf (
+                ngettext ("%'d hour", "%'d hours", hours),
+                ngettext ("%'d minute", "%'d minutes", minutes)
+            ).printf (hours, minutes);
+        }
+
+        time_unit = hours;
+        return ngettext ("approximately %'d hour", "approximately %'d hours", hours).printf (hours);
+    }
+
+    internal int get_max_name_length (GLib.File file_dir) {
+        if (!file_dir.has_uri_scheme ("file")) {
+            return -1;
+        }
+
+        var dir = file_dir.get_path ();
+        var max_path = Posix.pathconf (dir, Posix.PathConfName.PATH_MAX);
+        var max_name = Posix.pathconf (dir, Posix.PathConfName.NAME_MAX);
```

Comment:
Need to bump minimum version of `valac` in order to support `Posix.pathconf ()`.  On Hera, fails to compile with valac version 0.40.  These lines compile with valac version 0.48 but build fails due to faults in the zeitgeist.vapi.  Unless this can be fixed we will have to hold off merging this until Odin is released.

Reply:
done, Odin is now the target branch anyway :slightly_smiling_face: